src/llm/http_llm_calculator.cc

```diff
@@ -442,6 +488,7 @@ class HttpLLMCalculator : public CalculatorBase {
             nodeResources->notifyExecutorThread();
             this->streamer = std::make_shared<TextStreamer>(
                 nodeResources->cbPipe->get_tokenizer());
+            this->request->setEndpoint(endpoint);
```

Comment:
I think it would be better to move setEndpoint to line 458/460 so that we can use it in `parse` later on. What is more, for undefined endpoint (line 462) I would RET_CHECK it early, even before calling parse.